#!/usr/bin/env python
PACKAGE = "armcontrol_demo_pkg"
 
from dynamic_reconfigure.parameter_generator_catkin import *
 
gen = ParameterGenerator()

motor_state = gen.enum([gen.const("disable",    str_t,  "disable",  "电机失能，进入被动阻尼状态"),
                        gen.const("enable",     str_t,  "enable",   "电机使能，可以控制")],
                        "控制机械臂电机开启或关闭")

arm_mode = gen.enum([gen.const("arm_sleep",             str_t,	"arm_sleep",           "机械臂收拢到休眠位置，解除位置控制进入阻尼模式，适合长时间保持"),
                     gen.const("arm_control_demo",      str_t,  "arm_control_demo",       "机械臂控制指令不经过moveit规划,实现点动和拖动示教功能"),
                     gen.const("arm_control_moveit_fk",	str_t,	"arm_control_moveit_fk", "使用moveit进行正运动学规划,此时输入量为各关节角度"),
                     gen.const("arm_control_moveit_ik",	str_t,	"arm_control_moveit_ik", "使用moveit进行逆运动学规划,此时输入量为末端位姿")],
                     "控制机械臂处于何种控制模式")

drag_teachin = gen.enum([gen.const("false",	str_t,	"false",	"停止"),
                         gen.const("write",	str_t,	"write",	"写入"),
                         gen.const("read",	str_t,	"read", "读出")],
                        "拖动示教")

#设定可调节参数
gen.add("motor_state",      str_t,  0,  "控制机械臂电机开启或关闭",     "disable",      edit_method=motor_state)
gen.add("arm_mode",         str_t,  0,  "控制机械臂处于何种控制模式",   "arm_sleep",    edit_method=arm_mode)
gen.add("Emergency_Stop",   bool_t, 0,  "急停",     False)
gen.add("drag_teachin",     str_t,  0,  "拖动示教", "false",                        edit_method=drag_teachin)
gen.add("drag_teachin_name",str_t,  0,  "拖动示教存储文件名",  "/home/arm003/roboarm_ws/src/dm_motor/txt/drag_teachin_1.txt")

gen.add("joint1_angle", double_t, 0, "关节1角度", 0.0, -3.14, 3.14)
gen.add("joint2_angle", double_t, 0, "关节2角度", 0.0, -3.14, 3.14)
gen.add("joint3_angle", double_t, 0, "关节3角度", 0.0, -3.14, 3.14)
gen.add("joint4_angle", double_t, 0, "关节4角度", 0.0, -3.14, 3.14)
gen.add("joint5_angle", double_t, 0, "关节5角度", 0.0, -3.14, 3.14)
gen.add("joint6_angle", double_t, 0, "关节6角度", 0.0, -3.14, 3.14)

gen.add("arm_position_x", double_t, 0, "末端x坐标", 0.2, -0.5, 0.5)
gen.add("arm_position_y", double_t, 0, "末端y坐标", 0.2, -0.5, 0.5)
gen.add("arm_position_z", double_t, 0, "末端z坐标", 0.3, -0.6, 0.7)
gen.add("arm_orientation_roll",     double_t, 0, "末端姿态roll角",  0.0,    -3.14, 3.14)
gen.add("arm_orientation_pitch",    double_t, 0, "末端姿态pitch角", 1.57,   -3.14, 3.14)
gen.add("arm_orientation_yaw",      double_t, 0, "末端姿态yaw角度", 0.0,    -3.14, 3.14)

exit(gen.generate(PACKAGE,"armcontrol_demo_node","DynamicParam"))   
